src/radical/pilot/agent/scheduler/base.py

```diff
@@ -748,6 +748,8 @@ def _unschedule_completed(self):
             while not self._proc_term.is_set():
                 unit = self._queue_unsched.get(timeout=0.001)
                 to_unschedule.append(unit)
+                if len(to_unschedule) > 128:
+                    break
 
```

Comment:
Both the timeout and the threshold seem arbitrary. Something to try to fix with a different design? Meanwhile, should we expose them via (private) conf parameters?

Reply:
They are very arbitrary indeed, but the behaviour is very benign: the goal is to avoid corner cases: for the sleep, avoid no sleep (busy idle) and avoid sleep which introduces significant latencies (throughput).  Anything smaller than 0.01 is under our noise level and works ok for the latency, and anything larger than 0 is sufficient to avoid busy idle.

For the `unschedule` bulk, the corner case to avoid is waiting for too long to fill a bulk so that latencies add a up and negate the bulk optimization.  For the 0.001 sleep, 128 as bulk size results in a max added latency of about 0.1 second, which is one order of magnitude above our noise level again.

I'll add a comment :-)